File: packages/ios/Sources/Components/Slider/Slider.figma.swift
import SwiftUI
import Figma

struct KozmosSliderConnect: FigmaConnect {
    let component = KozmosSlider.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=80-473"

    @FigmaString("Label Text")
    var label: String = "Value"

    @FigmaEnum(
        "State",
        mapping: [
            "Default": "default",
            "Focus": "focus",
            "Disabled": "disabled"
        ]
    )
    var state: String = "default"

    var body: some View {
        VStack(alignment: .leading, spacing: KozmosDimensions.primitivesLayoutSpacing100) {
            Text(self.label)
            KozmosSlider(value: .constant(0.5))
                .disabled(self.state == "disabled")
        }
    }
}

File: packages/android/src/main/java/com/kozmos/components/Slider/Slider.figma.kt
package com.kozmos.components.slider

import androidx.compose.runtime.Composable
import com.figma.code.connect.Figma
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=80-473")
class KozmosSliderConnect {
    @FigmaProperty(FigmaType.Text, "Label Text")
    val label: String = "Value"

    @FigmaProperty(FigmaType.Enum, "State")
    val state: String = Figma.mapping(
        "Default" to "default",
        "Focus" to "focus",
        "Disabled" to "disabled"
    )

    @Composable
    fun ComponentExample() {
        KozmosSlider(
            value = 0.5f,
            onValueChange = {},
            enabled = state != "disabled"
        )
    }
}
